text/0000-a11y-routing.md

```diff
@@ -22,66 +22,134 @@ One of the challenges is that screen reader technology is all closed-source exce
+  clip: rect(0, 0, 0, 0);
+  white-space: nowrap;
+  border: 0;
+}
+```
+
+The addon would attempt to provide a sensible resolution for all involved:
+- the performance gains from `pushState` remain in place
+- users with assistive technology are informed that a page transition has occurred
+- the focus is reset to the message itself, which also resets the focus of the page (as is desired for the screen-reader user)
+- `aria-live` can remain available for things that genuinely need it (a page transition should not need it)
+
+Some [experimentation of this approach can be seen online](https://navigator-message-test-app.netlify.com/). In this example, the message _is_ shown for demonstration purposes, but in the final product, the message would not be shown.  
+
+What do users with screen-readers think? "It works better than I thought it might."
+
+### Solution #2 - Content Focus
+
+What about content-level focus? [Ember-self-focused](https://github.com/linkedin/self-focused/tree/master/packages/ember-self-focused) and [ember-a11y](https://github.com/ember-a11y/ember-a11y) are both examples of community-produced addons that attempt to provide a contextual focus, based on new page content. 
+
+After some research, we discovered that we could provide content-level focus by using (currently private) API to place focus around the content of the application. 
 
-For phase one, the guides should include information about skip links, since the application author will need to add those in themselves (just as they would for a static site). For phase two, we would explain that skip links are no longer needed because we have integrated that functionality in a more machine intelligent way. 
+For the template, we could take one of two approaches: 
 
-### Skip Links
-A skip link is an internal link at the beginning of a hypertext document that permits users to skip navigational material and quickly access the document's main content. Skip links are particularly useful for users who access a document with screen readers and users who rely on keyboards.
+1. wrap the `{{outlet}}` on application.hbs with an element with the tabindex and id attributes, like this:
 
-Directly inside the `body` element, place a skip link like this: 
+```hbs
+<main tabindex="-1" id="ember-primary-application-outlet">
+  {{outlet}}
+</main>
+```
 
-    <a href="#maincontent">Skip to main content</a> 
+2. change the application.hbs outlet so it is different from the usual `{{outlet}}`, and have it already include what we need:
 
-This allows the user with assistive technology to skip directly to the main content and is useful for not needing to repeatedly navigate through a navbar, for example.  
+```hbs
+{{application-outlet}}
+```
 
+In this way, if an application author did not want to use the `{{application-outlet}}`, for whatever reason (maybe they are nesting apps, or maybe they already have a focus solution) they could remove the `application-` and be left with a classic outlet. 
```

Comment:
I feel like the behavior in this scenario should be opt out, not opt in. Start with `outlet` that does the accessible thing, opting out requires changing things in the application. Presumably the person who knows how to opt out will also be more familiar with what they're doing.